backend/codingchallenge/api/views/applications.py

```diff
@@ -0,0 +1,105 @@
+import time
+import random
+
+# Authentication imports
+from rest_framework.permissions import IsAdminUser, IsAuthenticated
+
+# RESTapi imports
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+# import model
+from ..models import Application, Challenge
+
+# import serializer
+from ..serializers import ApplicationSerializer
+
+# import errorMessage class
+from . import errorMessage
+
+### endpoint: /api/admin/applications
+class AdminApplicationsView(APIView):
+    # grant permission only for admin user
+    permission_classes = [IsAdminUser]
+
+    name = "Admin Application View"
+    description = "handling all requests for applications as a admin"
+
+    # 1. List all
+    def get(self, request, *args, **kwargs):
+        '''
+        get Application with follwing columns:
+            applicationId,
+            operatingSystem,
+            programingLanguage,
+            expiry,
+            submission,
+            githubRepo,
+            status,
+            created,
+            modified
+        '''
+        applications = Application.objects
+        serializer = ApplicationSerializer(applications, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    # 2. Create
+    def post(self, request, *args, **kwargs):
+        '''
+        post Application with
+            required arguments:
+                applicationId,
+                applicantEmail
+
+            optional arguments:
+                challengeId
+                days
+        '''
+
+        try:
+            # random challenge
+            challengeId = random.choice(Challenge.objects.all()).id
+            # default 2 days time for start
+            days = 2
```

Comment:
Maybe better to set a (class) member variable for the default time.

Reply:
fixed in #63